src/main/scala/fi/vm/sade/hakurekisteri/integration/valintatulos/SijoitteluTulos.scala

```diff
@@ -127,7 +144,40 @@ object SijoitteluTulos {
       valintatulos.hakutoiveet
         .filter(h => h.valintatapajonoOid.nonEmpty)
         .map(h => (valintatulos.hakemusOid, h.hakukohdeOid) -> h.varasijanumero)
-        .toMap
+        .toMap,
+      valintatulos.hakutoiveet
+        .map(h =>
+          (valintatulos.hakemusOid, h.hakukohdeOid) ->
+            HyvaksymisenEhto(
+              ehdollisestiHyvaksyttavissa = h.ehdollisestiHyvaksyttavissa,
+              ehtoKoodi = h.ehdollisenHyvaksymisenEhtoKoodi,
+              ehtoFI = h.ehdollisenHyvaksymisenEhtoFI,
+              ehtoSV = h.ehdollisenHyvaksymisenEhtoSV,
+              ehtoEN = h.ehdollisenHyvaksymisenEhtoEN
+            )
+        )
+        .toMap,
+      valintatulos.hakutoiveet.collect {
+        case ValintaTulosHakutoive(
+              oid,
+              _,
+              _,
+              _,
+              _,
+              Some(valinnanAikaleima),
+              _,
+              _,
+              _,
+              _,
+              _,
+              _,
+              _,
+              _,
+              _,
+              _
+            ) =>
+          (valintatulos.hakemusOid, oid) -> valinnanAikaleima
+      }.toMap
```

Comment:
Olisikohan joku tämän tyylinen siistimpi, alaviivojen määrä vähän särähtää silmään :)
```suggestion
            valintatulos.hakutoiveet
        .map((ht: ValintaTulosHakutoive) => (valintatulos.hakemusOid, ht.hakukohdeOid) -> ht.hyvaksyttyJaJulkaistuDate.getOrElse(""))
        .filter(_._2.nonEmpty)
        .toMap
```
Ehkä tuota saisi vielä jollain vähän sievennettyä.

Reply:
Joo, tuo pattern matching -härvelin koko alkoi kieltämättä jo hirvittää editoidessakin - voisi siistiä nuo.